pagerduty/resource_pagerduty_slack_connection.go

```diff
@@ -0,0 +1,309 @@
+	slackConn := pagerduty.SlackConnection{
+		SourceID:         d.Get("source_id").(string),
+		SourceName:       d.Get("source_name").(string),
+		SourceType:       d.Get("source_type").(string),
+		ChannelID:        d.Get("channel_id").(string),
+		ChannelName:      d.Get("channel_name").(string),
+		WorkspaceID:      d.Get("workspace_id").(string),
+		NotificationType: d.Get("notification_type").(string),
+		Config:           expandConnectionConfig(d.Get("config").(interface{})),
+	}
+	return &slackConn, nil
+}
+
+func resourcePagerDutySlackConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	config := &pagerduty.Config{
+		Token:   os.Getenv("PAGERDUTY_USER_TOKEN"),
+		BaseURL: "https://app.pagerduty.com",
+	}
+	client, err := pagerduty.NewClient(config)
+	if err != nil {
+		return err
+	}
+
+	retryErr := resource.Retry(2*time.Minute, func() *resource.RetryError {
+
+		slackConn, err := buildSlackConnectionStruct(d)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		log.Printf("[INFO] Creating PagerDuty slack connection for source %s and slack channel %s", slackConn.SourceID, slackConn.ChannelID)
+
+		if slackConn, _, err = client.SlackConnections.Create(slackConn.WorkspaceID, slackConn); err != nil {
+			return resource.RetryableError(err)
+		} else if slackConn != nil {
+			d.SetId(slackConn.ID)
+			d.Set("workspace_id", slackConn.WorkspaceID)
+		}
+		return nil
+	})
+	if retryErr != nil {
+		time.Sleep(2 * time.Second)
+		return retryErr
+	}
+	return resourcePagerDutySlackConnectionRead(d, meta)
+}
+
+func resourcePagerDutySlackConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	config := &pagerduty.Config{
+		Token:   os.Getenv("PAGERDUTY_USER_TOKEN"),
+		BaseURL: "https://app.pagerduty.com",
+	}
+	client, err := pagerduty.NewClient(config)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Reading PagerDuty slack connection %s", d.Id())
+
+	workspaceID := d.Get("workspace_id").(string)
+	log.Printf("[DEBUG] Read Slack Connection: workspace_id %s", workspaceID)
+
+	retryErr := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		if slackConn, _, err := client.SlackConnections.Get(workspaceID, d.Id()); err != nil {
+			return resource.RetryableError(err)
+		} else if slackConn != nil {
+			d.Set("source_id", slackConn.SourceID)
+			d.Set("source_name", slackConn.SourceName)
+			d.Set("source_type", slackConn.SourceType)
+			d.Set("channel_id", slackConn.ChannelID)
+			d.Set("channel_name", slackConn.ChannelName)
+			d.Set("notification_type", slackConn.NotificationType)
+			d.Set("config", flattenConnectionConfig(slackConn.Config))
+		}
+		return nil
+	})
+
+	if retryErr != nil {
+		time.Sleep(2 * time.Second)
+		return retryErr
+	}
+
+	return nil
+}
+
+func resourcePagerDutySlackConnectionUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := &pagerduty.Config{
+		Token:   os.Getenv("PAGERDUTY_USER_TOKEN"),
+		BaseURL: "https://app.pagerduty.com",
```

Comment:
Do we want to make this a const? 🙂

Reply:
Will it work with the EU service region ?
I can see the API URL is different depending on using US or EU region
Might be the case also for app.pagerduty.com ?

Reply:
Good point @jbfavre! I confirmed with PagerDuty engineering that is the case. 🤔 Will need to figure out a better solution for setting baseURL.